### DISCUSSION TITLE
Building termux-packages dependencies with Github Actions of another repository

### DISCUSSION THREAD

**Discussion by twaik** [termux/termux-packages#14606] [General]
Hi. I need to build libx11 for my app (to have it in `/libs` inside `.apk`). I know I can use [these workflows](https://github.com/termux/termux-packages/tree/master/.github/workflows) but I am not sure if I should. I have some questions about it.
1. What workflow should I use? `packages.yml`? `package_updates.yml`? Or I should define my own? How does it work?
2. Will workflow rebuild all packages or it will unpack prebuilt dev packages from apt repos and their dependencies?
3. I think I need to build libX11 and all dependencies for all needed archs before I start Gradle build. Is it a problem? I see `packages.yml` uses some matrix in strategy section.
3. What about building on local pc? 

Thank you guys.

**Comment by twaik** [comment]
@landfillbaby ? @Grimler91 ? @thunder-coding ? @licy183 ?

**Comment by licy183** [comment]
Hello twaik, sorry for my late reply. I'm working on my final exam these days.
In my point of view, you should write your own one. The workflows in termux-packages cannot work out-of-box for your need. They pick up the commit messages from `termux-packages` repository and find which packages should be built, but `libx11` (and its dependencies) are the libraries you may want to build.
It seems better for you to download the prebuilt debs from the apt repository and unpack them. I think this won't be too difficult to do in a workflow.
Of cource, you can also build `libx11`, `libxkbcommon` and `libandroid-support` from source. If you really need to build it, just do the following steps in Action workflow. 
1. clone `termux-packages`
2. run `./build-package.sh -I libx11`
3. unpack the deb files and copy the libx11 and  its dependencies.

Matrix in workflow won't be a problem. In the workflow of termux-packages, it is used to run the building process in parallel and speed up the building process. You can use archives to transfer data from different jobs in workflow.
You can also build these libraries in your local pc. Just run `./scripts/run-docker.sh ./build-package.sh -I libx11`.
